diff -U1 -r db-6.2.23/src/dbinc/win_db.h db-patched/src/dbinc/win_db.h
--- db-6.2.23/src/dbinc/win_db.h	2016-03-28 19:45:54.000000000 +0000
+++ db-patched/src/dbinc/win_db.h	2019-07-21 09:03:54.399513387 +0000
@@ -53,3 +53,3 @@
 #ifndef DB_WINCE
-#include <WinIoCtl.h>
+#include <winioctl.h>
 #endif
diff -U2 -r db-6.2.23/dist/Makefile.in db-patched/dist/Makefile.in
--- db-6.2.23/dist/Makefile.in	2016-03-28 19:45:48.000000000 +0000
+++ db-patched/dist/Makefile.in	2019-07-21 09:51:02.281327824 +0000
@@ -852,7 +852,7 @@
 UTIL_PROGS=\
 	@ADDITIONAL_PROGS@ \
-	db_archive db_checkpoint db_deadlock db_dump \
-	db_hotbackup db_load db_log_verify db_printlog db_recover \
-	db_replicate db_stat db_tuner db_upgrade db_verify
+	db_archive@EXEEXT@ db_checkpoint@EXEEXT@ db_deadlock@EXEEXT@ db_dump@EXEEXT@ \
+	db_hotbackup@EXEEXT@ db_load@EXEEXT@ db_log_verify@EXEEXT@ db_printlog@EXEEXT@ db_recover@EXEEXT@ \
+	db_replicate@EXEEXT@ db_stat@EXEEXT@ db_tuner@EXEEXT@ db_upgrade@EXEEXT@ db_verify@EXEEXT@
 
 ##################################################
@@ -1064,49 +1064,49 @@
 # Utilities
 ##################################################
-db_archive: db_archive@o@ util_sig@o@ $(DEF_LIB)
+db_archive@EXEEXT@: db_archive@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_archive@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_checkpoint: db_checkpoint@o@ util_log@o@ util_sig@o@ $(DEF_LIB)
+db_checkpoint@EXEEXT@: db_checkpoint@o@ util_log@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_checkpoint@o@ util_log@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_deadlock: db_deadlock@o@ util_log@o@ util_sig@o@ $(DEF_LIB)
+db_deadlock@EXEEXT@: db_deadlock@o@ util_log@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_deadlock@o@ util_log@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_dump: db_dump@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
+db_dump@EXEEXT@: db_dump@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_dump@o@ util_cache@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_dump185: db_dump185@o@ @REPLACEMENT_OBJS@
+db_dump185@EXEEXT@: db_dump185@o@ @REPLACEMENT_OBJS@
 	$(CCLINK) -o $@ $(LDFLAGS) db_dump185@o@ @REPLACEMENT_OBJS@ $(DB185LIB)
 	$(POSTLINK) $@
 
-db_hotbackup: db_hotbackup@o@ util_sig@o@ $(DEF_LIB)
+db_hotbackup@EXEEXT@: db_hotbackup@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_hotbackup@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_load: db_load@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
+db_load@EXEEXT@: db_load@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_load@o@ util_cache@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_printlog: db_printlog@o@ @PRINTLOG_OBJS@ util_sig@o@ $(DEF_LIB)
+db_printlog@EXEEXT@: db_printlog@o@ @PRINTLOG_OBJS@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_printlog@o@ @PRINTLOG_OBJS@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_recover: db_recover@o@ util_sig@o@ $(DEF_LIB)
+db_recover@EXEEXT@: db_recover@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_recover@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_replicate: db_replicate@o@ util_sig@o@ $(DEF_LIB)
+db_replicate@EXEEXT@: db_replicate@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_replicate@o@ util_log@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
@@ -1118,29 +1118,29 @@
           generation_utils@o@ generate_verification@o@ hint_comment@o@
 
-db_sql_codegen: $(DBSQL_CODEGEN_OBJS) $(DEF_LIB)
+db_sql_codegen@EXEEXT@: $(DBSQL_CODEGEN_OBJS) $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) $(DBSQL_CODEGEN_OBJS) $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_stat: db_stat@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
+db_stat@EXEEXT@: db_stat@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_stat@o@ util_cache@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_tuner: db_tuner@o@ util_sig@o@ $(DEF_LIB)
+db_tuner@EXEEXT@: db_tuner@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_tuner@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_upgrade: db_upgrade@o@ util_sig@o@ $(DEF_LIB)
+db_upgrade@EXEEXT@: db_upgrade@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_upgrade@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_verify: db_verify@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
+db_verify@EXEEXT@: db_verify@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_verify@o@ util_cache@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
 	$(POSTLINK) $@
 
-db_log_verify: db_log_verify@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
+db_log_verify@EXEEXT@: db_log_verify@o@ util_cache@o@ util_sig@o@ $(DEF_LIB)
 	$(CCLINK) -o $@ $(LDFLAGS) \
 	    db_log_verify@o@ util_cache@o@ util_sig@o@ $(DEF_LIB) $(LIBS)
@@ -1214,5 +1214,4 @@
 	    $(CHMOD) $(dmode) $(DESTDIR)$(bindir))
 	@for i in $(UTIL_PROGS); do \
-		test -f $$i.exe && i=$$i.exe; \
 		e=`echo $$i | $(SED) '$(transform)'`; \
 		$(RM) $(DESTDIR)$(bindir)/$$e; \
@@ -1225,5 +1224,4 @@
 	@(cd $(DESTDIR)$(bindir); for i in $(UTIL_PROGS); do \
 		i=`echo $$i | $(SED) '$(transform)'`; \
-		test -f $$i.exe && i=$$i.exe; \
 		$(RM) $$i; \
 	done)
@@ -1293,5 +1291,5 @@
 	$(RM) -r $(C_OBJS) $(CUTEST_OBJS)
 	$(RM) -r $(CXX_OBJS) $(JAVA_OBJS) $(SQL_OBJS) $(SQL_OBJS) $(STL_OBJS)
-	$(RM) -r $(TCL_OBJS) $(UTIL_PROGS) *.exe $(CLEAN_LIST)
+	$(RM) -r $(TCL_OBJS) $(UTIL_PROGS) $(CLEAN_LIST)
 	$(RM) -r $(JAVA_CLASSTOP) $(JAVA_EXCLASSTOP) $(JAVA_DEMOCLASSTOP)
 	$(RM) -r $(DB_STL_TEST_OBJS) $(TEST_MICRO_OBJS)
